test/system/src/main/java/io/pravega/test/system/framework/services/kubernetes/K8SequentialExecutor.java

```diff
@@ -0,0 +1,88 @@
+                                                   .thenCompose(v -> client.waitUntilPodCompletes(NAMESPACE, podName))
+                                                   .handle((s, t) -> {
+                                                       if (t == null) {
+                                                           log.info("Test execution completed with status {}", s);
+                                                           client.saveLogs(pod, "./build/test-results/" +podName); //save test log.
+                                                           if (s.getExitCode() != 0) {
+                                                               log.error("Test {}#{} failed. Details: {}", className, methodName, s);
+                                                               throw new AssertionError(methodName + " test failed.");
+                                                           } else {
+                                                               return null;
+                                                           }
+
+                                                       } else {
+                                                           throw new CompletionException("Error while invoking the test " + podName, t);
+                                                       }
+                                                   });
+
+        testFuture.thenRun(client::close); // close the client.
+        return testFuture;
+    }
+
+    private V1Pod getTestPod(String className, String methodName, String podName) {
+        String repoUrl = System.getProperty("repoUrl");
+        String testVersion = System.getProperty("testVersion");
+        return new V1PodBuilder()
+                .withNewMetadata().withName(podName).withNamespace(NAMESPACE).withLabels(ImmutableMap.of("POD_NAME", podName)).endMetadata()
+                .withNewSpec().addNewContainer()
+                .withName(podName) // container name is same as that of the pod.
+                .withImage("openjdk:8-jre-alpine")
+                .withImagePullPolicy("IfNotPresent")
+                .withCommand("/bin/sh")
+                .withArgs("-c",
+                          "wget " + repoUrl + "/io/pravega/pravega-test-system/" + testVersion + "/pravega-test-system-" + testVersion +".jar && "
+                                  + "echo \"download of system test jar complete\" && "
+                                  + "java -DexecType=K8 -cp ./pravega-test-system-" + testVersion + ".jar io.pravega.test.system.SingleJUnitTestRunner "
+                                  + className + "#" +methodName /*+ " > server.log 2>&1 */ + "; exit $?")
+                .endContainer()
+                .withRestartPolicy("Never")
+                .endSpec().build();
+    }
+
+    @Override
+    public void stopTestExecution() {
+        throw new NotImplementedException("Not implemented for Kubernetes based tests");
```

Comment:
What's the plan here? Is it not doable at all or the idea is to do it later?

Reply:
Currently the `stopTestExecution` api is never invoked by the test framework. (it is not implemented for dcos / marathon or kubernetes based test executors.). If required we can enable it for all types of executors in future.